src/swim/swim.c

```diff
@@ -41,7 +41,72 @@
 #include "swim_internal.h"
 #include <assert.h>
 
-static uint64_t swim_ping_timeout = SWIM_PING_TIMEOUT;
+static uint64_t swim_prot_period_len;
+static uint64_t swim_suspect_timeout;
+static uint64_t swim_ping_timeout;
+
+static inline uint64_t
+swim_prot_period_len_default(void)
+{
+	unsigned int val = SWIM_PROTOCOL_PERIOD_LEN;
+
+	d_getenv_int("SWIM_PROTOCOL_PERIOD_LEN", &val);
+	return val;
+}
+
+static inline uint64_t
+swim_suspect_timeout_default(void)
+{
+	unsigned int val = SWIM_SUSPECT_TIMEOUT;
+
+	d_getenv_int("SWIM_SUSPECT_TIMEOUT", &val);
+	return val;
+}
+
+static inline uint64_t
+swim_ping_timeout_default(void)
+{
+	unsigned int val = SWIM_PING_TIMEOUT;
+
+	d_getenv_int("SWIM_PING_TIMEOUT", &val);
+	return val;
+}
+
+void
+swim_period_set(uint64_t val)
+{
+	swim_prot_period_len = val;
```

Comment:
on all _set() calls we probably want to do D_DEBUG() to dump value into log file

Reply:
ok